semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/postgres-connector.md

```diff
@@ -62,91 +86,420 @@ using Npgsql;
+in a connection pool, the collection will create one using the `PostgresSettings` class.
+
+```python
+from semantic_kernel.connectors.memory.Postgres import PostgresCollection
+
+collection = PostgresCollection(collection_name="skhotels", data_model_type=Hotel)
+with collection: # This will create a connection pool using PostgresSettings
+    ...
+```
+
+## Data mapping
+
+The Postgres connector provides a default mapper when mapping data from the data model to storage.
+The default mapper uses the model annotations or record definition to determine the type of each property and to map the model
+into a `dict` that can be serialized to Postgres rows.
+
+- The data model property annotated as a key will be mapped to the PRIMARY KEY in the Postgres table.
+- The data model properties annotated as data will be mapped to a table column in Postgres.
+- The data model properties annotated as vectors will be mapped to a table column that has the pgvector `VECTOR` type in Postgres.
+
+```python
+from pydantic import BaseModel
+
+from semantic_kernel.data import (
+    DistanceFunction,
+    IndexKind,
+    VectorStoreRecordDataField,
+    VectorStoreRecordKeyField,
+    VectorStoreRecordVectorField,
+    vectorstoremodel,
+)
+
+@vectorstoremodel
+class Hotel(BaseModel):
+    hotel_id: Annotated[int, VectorStoreRecordKeyField()]
+    hotel_name: Annotated[str, VectorStoreRecordDataField()]
+    hotel_description: Annotated[str, VectorStoreRecordDataField(has_embedding=True, embedding_property_name="description_embedding")]
+    description_embedding: Annotated[
+        list[float] | None,
+        VectorStoreRecordVectorField(
+            index_kind=IndexKind.HNSW,
+            dimensions=4,
+            distance_function=DistanceFunction.COSINE_SIMILARITY,
+        ),
+    ] = None
+```
+
+```sql
+CREATE TABLE Hotels (
+    hotel_id INTEGER PRIMARY KEY,
+    hotel_name TEXT,
+    hotel_description TEXT,
+    description_embedding VECTOR(4)
+);
+```
+
+## Using with Entra Authentication
+
+Azure Database for PostgreSQL provides the ability to connect to your database using [Entra authentication](/azure/postgresql/flexible-server/concepts-azure-ad-authentication).
+This removes the need to store a username and password in your connection string.
+To use Entra authentication for an Azure DB for PostgreSQL database, you can use the following custom AsyncConnection class:
+
+```python
+import base64
+import json
+import logging
+from functools import lru_cache
+
+from azure.core.credentials import TokenCredential
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.identity import DefaultAzureCredential
+from psycopg import AsyncConnection
+
+AZURE_DB_FOR_POSTGRES_SCOPE = "https://ossrdbms-aad.database.windows.net/.default"
+
+logger = logging.getLogger(__name__)
+
+
+async def get_entra_token_async(credential: AsyncTokenCredential) -> str:
```

Comment:
in `semantic_kernel/utils/authentication/entra_id_authentication.py` we have prebuilt function for this, just pass the endpoint into that!

Reply:
That function is helpful, but there's not an async version and doesn't take in a credential as an option so always uses `DefaultAzureCredential`...passing in a credential is useful e.g. when you are operating in a service pod on AKS that has a workload identity, and you don't want to iterate through the possible credential paths because you only want a single credential mechanism.

I could contribute a PR that adds the credential argument and async version if you think that would be useful.

Reply:
that would be very useful, I'm very surprised there is no async in that one... and I like the idea of passing specific creds!

Reply:
Ok, put that on the TODO list. Looks like there's implications for the classes that use it, e.g. allowing credentials to be specificed in `AzureOpenAIConfigBase`.Do you think that should be blocking to this PR? It would be great to get this into the docs now that the postgres vectorstore feature is published. I can update the docs once the utils change gets merged and released.